packages/flutter/lib/src/material/tab_controller.dart

```diff
@@ -276,7 +281,9 @@ class TabController extends ChangeNotifier {
 
   @override
   void dispose() {
-    _animationController?.dispose();
+    if (_disposeAnimationController) {
```

Comment:
This is also a strange pattern. Either the TabController owns the AnimationController (then it needs to be disposed here) or it doesn't own it and should never dispose it.

Reply:
Added doc comment. Does it help?

Reply:
The pattern is still strange. How about: copyWith will move the animationController to the new TabController and null out _animationController on the old one (this should be added to the documentation of _copyWith). Then, instead of doing an internal `dispose` call in `_copyWith` whoever calls `_copyWith` should follow up with a dispose call if they no longer need the old controller.

Reply:
I removed the flag and nulled controller.

It is strange for me though to declare that whoever called _copyWith must call dispose. It is easy to miss. If it is always this way, why copyWith cannot do it?